scripts/remove-repo-license.js

```diff
@@ -9,40 +9,76 @@ function removeRepoLicense() {
   const noticesFile = './THIRD-PARTY-NOTICES';
 
   const noticesFileContents = fs.readFileSync(noticesFile, 'utf8');
-  const noticesWithoutRepoLicense = getNoticesWithoutReactSiteStarterLicense(noticesFileContents);
+  const noticesWithoutRepoLicense = getNoticesWithoutRepoLicense(noticesFileContents);
```

Comment:
hmm I looked into it and it seemed a little hard to directly use their api unless we copy/modify some parts. Mainly, we can use `getProjectLicenses` from their api to get an array of ILicense (license content + dependencies), but we would have to deal with formatting the licenses and constructing the whole file, unlike the `getProjectLicensesInternal` use by `getLicenseFileText` that returns array of License instances with format functions. Those stuff are not part of the public interface, we could copy getProjectLicensesInternal function and License class file, and modify getLicenseFileText to suit our use case. But that's extracting a lot of things from this library and make it hard to update version unless we just maintain this copied files ourselves.

Reply:
`getProjectLicenses` is just a pass through to `getProjectLicensesInternal`.
I think both having a remove-repo-license and writing our own `getLicenseFileText` with filtering have cons.

With remove-repo-license we're assuming implementation details like the exact text and format used by the package. If we update generate-license-file, and these change, our license generation may fail to generate correct license, and could even fail silently and create garbled license files (though hopefully this would be caught in PR). I'm being a stickler about this because we've already had one surprise of "Oh, we didn't know that generate-license-file would collapse shared licenses".

Writing our own `getLicenseFileText` doesn't assume any details of the generate-license file API. But it does force us to take responsibility for certain decisions, like the suffix and footer, even if we just copy paste those. On the other hand, we're already happy with the licenses as they are today. With that in mind, all using the API directly is doing is freezing that formatting in place.

We could also post a feature request in the repo to export some things like the suffix, footer, and `getLineEnding`, or even just add a flag to the CLI to not include your current package's license.

Reply:
Not sure what the rest of the team thinks but I'm in favor of forking `getLicenseFileText` and adding a tiny bit of extra logic to filter out the current package

Reply:
I sent a feature request to the library's github repo and it seems like they are working on the solution. Once that's publish, I will update this pr!